examples/src/threejs/utils/renderer-debug-widget.ts

```diff
@@ -0,0 +1,275 @@
+/*!
+ * Copyright 2019 Cognite AS
+ */
+
+import * as THREE from 'three';
+import dat from 'dat.gui';
+
+export type RenderFilter = {
+  renderQuads: boolean;
+  renderPrimitives: boolean;
+  renderTriangleMeshes: boolean;
+  renderInstancedMeshes: boolean;
+};
+const everythingRenderFilter: RenderFilter = {
+  renderQuads: true,
+  renderPrimitives: true,
+  renderTriangleMeshes: true,
+  renderInstancedMeshes: true
+};
+
+export enum RenderMode {
+  WhenNecessary = 'WhenNecessary',
+  DisableRendering = 'DisableRendering',
+  AlwaysRender = 'AlwaysRender'
+}
+
+export type RenderOptions = {
+  suspendLoading: boolean;
+  renderMode: RenderMode;
+  renderFilter: RenderFilter;
+};
+
+export function createDefaultRenderOptions() {
+  return {
+    suspendLoading: false,
+    renderMode: RenderMode.WhenNecessary,
+    renderFilter: everythingRenderFilter
+  };
+}
+
+function createEmptySceneInfo() {
+  return {
+    sectors: {
+      count: 0,
+      withMeshesCount: 0
+    },
+    primitives: {
+      meshCount: 0,
+      instanceCount: 0,
+      templateTriangleCount: 0
+    },
+    triangleMeshes: {
+      meshCount: 0,
+      triangleCount: 0
+    },
+    instanceMeshes: {
+      meshCount: 0,
+      instanceCount: 0,
+      avgInstancesPerMesh: 0.0,
+      templateTriangleCount: 0
+    },
+    quads: {
+      meshCount: 0,
+      quadCount: 0,
+      templateTriangleCount: 0
+    },
+    distinctMaterialCount: 0,
+    // Stuff to compute FPS
+    lastUpdate: {
+      timestamp: Date.now(),
+      frame: 0
+    },
+    fps: 0
+  };
+}
+type SceneInfo = ReturnType<typeof createEmptySceneInfo>;
+
+export function createRendererDebugWidget(
+  renderer: THREE.WebGLRenderer,
+  scene: THREE.Object3D,
+  parentGui?: dat.GUI,
+  intervalMs: number = 100
+): RenderOptions {
+  const gui = parentGui || new dat.GUI();
```

Comment:
Will this automatically appear on screen? If not, I think `gui` needs to be returned from this function.

Reply:
You can also make `parentGui` required instead. I do not think you use the optional version anywhere currently.